lib/jxl/frame_header.cc

```diff
@@ -397,10 +397,14 @@ Status FrameHeader::VisitFields(Visitor* JXL_RESTRICT visitor) {
     } else if (visitor->Conditional(frame_type == FrameType::kReferenceOnly)) {
       JXL_QUIET_RETURN_IF_ERROR(
           visitor->Bool(true, &save_before_color_transform));
+      size_t xsize = custom_size_or_origin ? frame_size.xsize
+                                           : nonserialized_metadata->xsize();
+      size_t ysize = custom_size_or_origin ? frame_size.ysize
+                                           : nonserialized_metadata->ysize();
       if (!save_before_color_transform &&
-          (frame_size.xsize < nonserialized_metadata->xsize() ||
-           frame_size.ysize < nonserialized_metadata->ysize() ||
-           frame_origin.x0 != 0 || frame_origin.y0 != 0)) {
+          (xsize < nonserialized_metadata->xsize() ||
+           ysize < nonserialized_metadata->ysize() || frame_origin.x0 != 0 ||
+           frame_origin.y0 != 0)) {
         return JXL_FAILURE(
```

Comment:
Shouldn't the error message below also use xsize/ysize?

Reply:
yup